src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -357,7 +361,17 @@ BlockIO InterpreterCreateQuery::createDatabase(ASTCreateQuery & create)
                         "Enable allow_experimental_database_materialized_postgresql to use it");
     }
 
-    bool need_write_metadata = !create.attach || !fs::exists(metadata_file_path);
+    bool need_write_metadata = !create.attach || !db_disk->existsFile(metadata_file_path);
+
+    LOG_DEBUG(
+        getLogger("InterpreterCreateQuery"),
+        "InterpreterCreateQuery metadata_file_path {}, create.attach {}, existsFile {}, need_write_metadata {}",
+        metadata_file_path,
+        create.attach,
+        db_disk->existsFile(metadata_file_path),
```

Comment:
This existsFile could be heavyweight for non local disks, it this log message still needed?

Reply:
Let me remove it.